kernel/first_application/src/lib.rs

```diff
@@ -28,7 +28,8 @@ use path::Path;
 
 /// See the crate-level docs and this crate's `Cargo.toml` for more.
 const FIRST_APPLICATION_CRATE_NAME: &str = {
-    #[cfg(target_arch = "x86_64")] { "shell-" }
+    #[cfg(all(target_arch = "x86_64", qemu_test))] { "qemu_test-" }
+    #[cfg(all(target_arch = "x86_64", not(qemu_test)))] { "shell-" }
```

Comment:
as in my other comment, i'd vote to use cargo features here, which would enable you to conditionally depend on your new `qemu_test` crate in `first_application/Cargo.toml`.

I'm also trying to gradually migrate the repo to using cargo features exclusively instead of rustc cfg (where possible), mostly for that reason of visibility at the manifest level.